docs/proposals/34-observability2.0-opentelemtry-operator-and-collectors.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Introducing OpenTelemetry Operator and Collectors in Shoot Control Planes
+gep-number: 34
+creation-date: 2025-04-12
+status: implementable
+authors:
+- "@nickytd"
+- "@rrhubenov"
+reviewers:
+-
+---
+
+# GEP-[34]: Introducing OpenTelemetry Operator and Collectors in Shoot Control Planes
+
+## Table of Contents
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+- [Alternatives](#alternatives)
+
+---
+
+## Summary
+
+This proposal introduces the OpenTelemetry operator and deploys OpenTelemetry collectors within the shoot control plane namespaces of Gardener-managed clusters. Building on the foundation laid by [GEP-19](https://github.com/gardener/gardener/blob/master/docs/proposals/19-migrating-observability-stack-to-operators.md) (migration to Prometheus and Fluent-bit operators) and the [Observability 2.0](https://github.com/gardener/logging/blob/master/docs/observability-2.0/Observability%202.0.md) vision, this GEP advances Gardener's observability stack by adopting OpenTelemetry standards. The initiative aims to layout the foundation for building a more cohesive and interoperable observability framework. This GEP takes the first step by introducing OpenTelemetry for logging, aligning with the [Observability 2.0](https://github.com/gardener/logging/blob/master/docs/observability-2.0/Observability%202.0.md) vision, with future enhancements planned to include metrics and optionally traces.
+
+---
+
+## Motivation
+
+Gardener's current observability stack, while improved by the operator-based approach in [GEP-19](https://github.com/gardener/gardener/blob/master/docs/proposals/19-migrating-observability-stack-to-operators.md), still relies on vendor specific format and protocols for collecting, processing and storing observability signals. This brings challenges in extending both components and scenarios used to process these signals and enforces lock-in integration scenarios with external consumers. By introducing the OpenTelemetry operator and collectors, this proposal addresses these challenges, aligning Gardener with industry trends and preparing it for future observability enhancements, such as unified visualization and tracing support.
+
+### Goals
+
+- **Deploy the OpenTelemetry Operator on Seed Clusters:** Install and configure the OpenTelemetry operator to manage otel-collector instances across shoot control plane namespaces.
```

Comment:
I guess we should add a goal to also deploy it in the garden runtime cluster?

Reply:
On a later phase that can achieved using the documented mechanisms for deploying components on runtime cluster as well. This is out of scope for the first iteration.